File: testers/avro/java/generated-and-checked-in/com/example/events/consumer/TreeNodeHandler.java
package com.example.events.consumer;

import com.example.events.TreeNode;
import com.example.events.header.StandardHeaders;

/** Handler interface for tree-node topic events */
public interface TreeNodeHandler {
  /** Handle a message from the topic */
  void handle(String key, TreeNode value, StandardHeaders headers);
}

File: testers/avro/scala/generated-and-checked-in/com/example/events/consumer/TreeNodeHandler.scala
package com.example.events.consumer

import com.example.events.TreeNode
import com.example.events.header.StandardHeaders

/** Handler interface for tree-node topic events */
trait TreeNodeHandler {
  /** Handle a message from the topic */
  def handle(
    key: String,
    value: TreeNode,
    headers: StandardHeaders
  ): Unit
}